### PR TITLE
Auto user hierarchy

### PR DESCRIPTION
**Pull request by eladex**
- User's hierarchy created automatically using the direct group.
- Tests fix: all the tests that create a person must create a group before (because the create function looks 
 for an existing group). this is done by adding "beforeEach" hook for the relevant tests.
- domainUser repo export is done in the same way as the other repo's

### CODE REVIEW

File: src/person/person.spec.ts [hunk 1]
```diff
@@ -21,6 +21,10 @@ chai.use(require('chai-http'));
 
 const dbIdExample = ['5b50a76713ddf90af494de32', '5b56e5ca07f0de0f38110b9c', '5b50a76713ddf90af494de33', '5b50a76713ddf90af494de34','5b50a76713ddf90af494de35','5b50a76713ddf90af494de36', '5b50a76713ddf90af494de37'];
```
Comment (DavidHeymann) [inline src/person/person.spec.ts]: Delete dbIdExample whether not used

File: src/person/person.spec.ts [hunk 2]
```diff
@@ -672,153 +674,6 @@ describe('Persons', () => {
 
 });
 
-async function bigTree() {
```
Comment (DavidHeymann) [inline src/person/person.spec.ts]: Why did you delete it?

File: src/person/person.spec.ts [hunk 3]
```diff
@@ -29,62 +33,60 @@ const personExamples: IPerson[] = [
...
-    hierarchy: ['www', 'microsoft', 'github'],
     job: 'Programmer',
     serviceType: SERVICE_TYPE[0],
-    directGroup: dbIdExample[3],
   },
 ];
 
 describe('Persons', () => {
+  // create OG to link with each person.
+  beforeEach(async () => {
```
Comment (DavidHeymann) [inline src/person/person.spec.ts]: Is there a way to not repeat the code twice. For example put the function in another file and use it in both places that need it

File: src/person/person.spec.ts (hunk 1 above)
Comment (eladex) [reply src/person/person.spec.ts]: It is used by the "responsibilityLocation" field. apparently, we do not check if this field refers to an existing group...
